pkg/function/interface.go

```diff
@@ -38,40 +38,40 @@ type Client interface {
 // MapHandler is the interface of map function implementation.
 type MapHandler interface {
 	// HandleDo is the function to process each coming message.
-	HandleDo(ctx context.Context, key string, datum Datum) Messages
+	HandleDo(ctx context.Context, key []string, datum Datum) Messages
 }
 
 // MapTHandler is the interface of mapT function implementation.
 type MapTHandler interface {
 	// HandleDo is the function to process each coming message.
-	HandleDo(ctx context.Context, key string, datum Datum) MessageTs
+	HandleDo(ctx context.Context, key []string, datum Datum) MessageTs
 }
 
 // ReduceHandler is the interface of reduce function implementation.
 type ReduceHandler interface {
-	HandleDo(ctx context.Context, key string, reduceCh <-chan Datum, md Metadata) Messages
+	HandleDo(ctx context.Context, key []string, reduceCh <-chan Datum, md Metadata) Messages
 }
 
 // MapFunc is utility type used to convert a HandleDo function to a MapHandler.
-type MapFunc func(ctx context.Context, key string, datum Datum) Messages
+type MapFunc func(ctx context.Context, key []string, datum Datum) Messages
 
 // HandleDo implements the function of map function.
-func (mf MapFunc) HandleDo(ctx context.Context, key string, datum Datum) Messages {
+func (mf MapFunc) HandleDo(ctx context.Context, key []string, datum Datum) Messages {
 	return mf(ctx, key, datum)
 }
 
 // MapTFunc is utility type used to convert a HandleDo function to a MapTHandler.
-type MapTFunc func(ctx context.Context, key string, datum Datum) MessageTs
+type MapTFunc func(ctx context.Context, key []string, datum Datum) MessageTs
```

Comment:
Since there's already API change, shall we get the `tags` supported in one shot?

Reply:
yeah we can track it for 0.8,